testsuite/standalone/src/test/java/org/wildfly/extras/creaper/commands/elytron/sasl/AddAggregateSaslServerFactoryOnlineTest.java

```diff
@@ -0,0 +1,207 @@
+                = new AddAggregateSaslServerFactory.Builder(TEST_SERVER_FACTORY_NAME)
+                .addSaslServerFactories(TEST_PROVIDER_SERVER_FACTORY_NAME,
+                        TEST_PROVIDER_SERVER_FACTORY_NAME2)
+                .build();
+
+        AddAggregateSaslServerFactory addAggregateSaslServerFactory2
+                = new AddAggregateSaslServerFactory.Builder(TEST_SERVER_FACTORY_NAME)
+                .addSaslServerFactories(TEST_PROVIDER_SERVER_FACTORY_NAME,
+                        TEST_PROVIDER_SERVER_FACTORY_NAME3)
+                .replaceExisting()
+                .build();
+
+        client.apply(addAggregateSaslServerFactory);
+        assertTrue("Aggregate sasl server factory should be created", ops.exists(TEST_SERVER_FACTORY_ADDRESS));
+        client.apply(addAggregateSaslServerFactory2);
+        assertTrue("Aggregate sasl server factory should be created", ops.exists(TEST_SERVER_FACTORY_ADDRESS));
+        // check whether it was really rewritten
+        checkAttribute(TEST_SERVER_FACTORY_ADDRESS, "sasl-server-factories[1]",
+                TEST_PROVIDER_SERVER_FACTORY_NAME3);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addAggregateSaslServerFactory_nullName() throws Exception {
+        new AddAggregateSaslServerFactory.Builder(null)
+                .addSaslServerFactories(TEST_PROVIDER_SERVER_FACTORY_NAME,
+                        TEST_PROVIDER_SERVER_FACTORY_NAME2)
+                .build();
+        fail("Creating command with null name should throw exception");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addAggregateSaslServerFactory_emptyName() throws Exception {
+        new AddAggregateSaslServerFactory.Builder("")
+                .addSaslServerFactories(TEST_PROVIDER_SERVER_FACTORY_NAME,
+                        TEST_PROVIDER_SERVER_FACTORY_NAME2)
+                .build();
+        fail("Creating command with empty name should throw exception");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addAggregateSaslServerFactory_nullSaslServerFactories() throws Exception {
+        new AddAggregateSaslServerFactory.Builder(TEST_SERVER_FACTORY_NAME)
+                .addSaslServerFactories(null)
+                .build();
+        fail("Creating command with null sasl-server-factories should throw exception");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addAggregateSaslServerFactory_emptySaslServerFactories() throws Exception {
+        new AddAggregateSaslServerFactory.Builder(TEST_SERVER_FACTORY_NAME)
+                .addSaslServerFactories("")
+                .build();
+        fail("Creating command with empty sasl-server-factories should throw exception");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void addAggregateSaslServerFactory_oneSaslServerFactories() throws Exception {
```

Comment:
`<grammar-nazi>`
~~oneSaslServerFactories~~ -> oneSaslServerFactory; the same applies also for `fail()` message
`</grammar-nazi>`